command/certificate/needsRenewal.go

```diff
@@ -0,0 +1,163 @@
+'''
+Check if certificate from a host has passed a percentage of its lifetime
+$ step certificate needs-renewal https://smallstep.com --expires-in 75%
+`,
+		Flags: []cli.Flag{
+			cli.StringFlag{
+				Name: "expires-in",
+				Usage: `Check if the certificate expires in given time duration
+				using <percent|duration>. For <percent>, must be followed by "%".
+				With <duration>, it is a sequence of decimal numbers, each with optional
+				fraction and a unit suffix, such as "300ms", "-1.5h" or "2h45m". Valid
+				time units are "ns", "us" (or "µs"), "ms", "s", "m", "h".`,
+			},
+		},
+	}
+}
+
+func needsRenewalAction(ctx *cli.Context) error {
+	if err := errs.NumberOfArguments(ctx, 1); err != nil {
+		return err
+	}
+	var (
+		crtFile    = ctx.Args().Get(0)
+		expiresIn  = ctx.String("expires-in")
+		roots      = ctx.String("roots")
+		serverName = ctx.String("servername")
+		cert       *x509.Certificate
+	)
+
+	if addr, isURL, err := trimURL(crtFile); err != nil {
+		return err
+	} else if isURL {
+		peerCertificates, err := getPeerCertificates(addr, serverName, roots, false)
+		if err != nil {
+			return errs.NewExitError(err, 255)
+		}
+		cert = peerCertificates[0]
+
+	} else {
+		crtBytes, err := ioutil.ReadFile(crtFile)
+		if err != nil {
+			return errs.NewExitError(err, 255)
+		}
+
+		var block *pem.Block
+		// The first certificate PEM in the file is our leaf Certificate.
+		// Any certificate after the first is added to the list of Intermediate
+		// certificates used for path validation.
+		for len(crtBytes) > 0 {
+			block, crtBytes = pem.Decode(crtBytes)
+			if block == nil {
+				return errs.NewExitError(errors.Errorf("%s contains an invalid PEM block", crtFile), 255)
+			}
+			if block.Type != "CERTIFICATE" {
+				continue
+			}
+			if cert == nil {
+				cert, err = x509.ParseCertificate(block.Bytes)
+				if err != nil {
+					return errs.NewExitError(errors.WithStack(err), 255)
+				}
+			}
+		}
+		if cert == nil {
+			return errs.NewExitError(errors.Errorf("%s contains no PEM certificate blocks", crtFile), 255)
+		}
+
+	}
+	var remainingValidity = time.Until(cert.NotAfter)
+	var totalValidity = cert.NotAfter.Sub(cert.NotBefore)
+	var percentUsed = (1 - remainingValidity.Minutes()/totalValidity.Minutes()) * 100
+
+	if expiresIn != "" {
+		if strings.Contains(expiresIn, "%") {
+			percentageInput, err := strconv.Atoi(strings.ReplaceAll(expiresIn, "%", ""))
+
+			if err != nil {
+				return errs.NewExitError(err, 255)
+			}
+			if percentageInput > 100 || percentageInput < 0 {
+				return errs.NewExitError(errors.Errorf("Percentage must be in range 0-100"), 255)
+			}
+
+			if percentageInput > int(percentUsed) {
+				return nil
+			}
+			os.Exit(1)
+
+		} else {
+			duration, err := time.ParseDuration(expiresIn)
+
+			if err != nil {
+				return errs.NewExitError(err, 255)
+			} else if duration.Minutes() > remainingValidity.Minutes() {
+				return nil
+			}
+			os.Exit(1)
+		}
+	} else {
+		if percentUsed >= 66 {
+			return nil
```

Comment:
It would be nice to combine this with the other percentage case so we don't copy the same conditional code.

Something like: 

```
for _, cert := range certs {
if expiredIn == "" || strings.Contains(expiresIn, "%") {
    var threshold int
    if expiresIn == "" {
        threshold = defaultPercentThreshold
    } else {
        threshold, err = strconv.Atoi(strings.TrimSuffix(expiresIn, "%"))
        if err ....
    }
    
    if threshold > 100 || threshold < 0 ...

    if int(percentUsed) >= threshold ...
        return nil
} else {
    			duration, err := time.ParseDuration(expiresIn)

			if err != nil {
				return errs.NewExitError(err, 255)
			} else if duration > remainingValidity {
				return nil
			}
}
}
// We only get here if the certs are not expiring, so return an error to that effect.
return errors.New("certificate is verdant")